src/Workflow/MoveJatsAssets.php

```diff
@@ -0,0 +1,164 @@
+use Libero\ContentApiBundle\Model\PutTask;
+use Psr\Http\Message\ResponseInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\File\MimeType\ExtensionGuesser;
+use Symfony\Component\Workflow\Event\Event;
+use function GuzzleHttp\Promise\each_limit_all;
+use function implode;
+use function in_array;
+use function Libero\ContentApiBundle\stream_hash;
+use function Libero\ContentStore\element_uri;
+use function Libero\ContentStore\parse_media_type;
+use function preg_match;
+use function sprintf;
+
+final class MoveJatsAssets implements EventSubscriberInterface
+{
+    private const HAS_MIMETYPE_ATTRIBUTE = [
+        'graphic',
+        'inline-graphic',
+        'inline-media',
+        'inline-supplementary-material',
+        'media',
+        'supplementary-material',
+    ];
+
+    private $client;
+    private $concurrency;
+    private $filesystem;
+    private $origin;
+    private $publicUri;
+
+    public function __construct(
+        string $origin,
+        string $publicUri,
+        FilesystemInterface $filesystem,
+        ClientInterface $client,
+        int $concurrency = 10
+    ) {
+        $this->origin = $origin;
+        $this->publicUri = $publicUri;
+        $this->filesystem = $filesystem;
+        $this->client = $client;
+        $this->concurrency = $concurrency;
+    }
+
+    public static function getSubscribedEvents() : array
+    {
+        return [
+            'workflow.libero.content_store.put.transition.manipulate' => 'onManipulate',
+        ];
+    }
+
+    public function onManipulate(Event $event) : void
+    {
+        /** @var PutTask $task */
+        $task = $event->getSubject();
+
+        each_limit_all(
+            $this->processAssets($this->findLinkedElements($task->getDocument()), $task),
+            $this->concurrency
+        )->wait();
+    }
+
+    /**
+     * @return iterable<Element>
+     */
+    private function findLinkedElements(Document $document) : iterable
+    {
+        $document->registerNamespace('jats', 'http://jats.nlm.nih.gov');
+        $document->registerNamespace('xlink', 'http://www.w3.org/1999/xlink');
+
+        yield from $document('//jats:article//jats:*[@xlink:href]');
+    }
+
+    private function processAssets(iterable $elements, PutTask $task) : iterable
+    {
+        foreach ($elements as $element) {
+            if ($this->shouldProcess($element)) {
+                yield $this->processAsset($element, $task);
+            }
+        }
+    }
+
+    private function processAsset(Element $asset, PutTask $task) : PromiseInterface
+    {
+        return new Coroutine(
+            function () use ($asset, $task) : iterable {
+                /** @var ResponseInterface $response */
+                $response = yield $this->client->requestAsync('GET', element_uri($asset));
+
+                $contentType = parse_media_type($response->getHeaderLine('Content-Type'));
```

Comment:
where does the specification "locations described in `@xlink:href` must have a `Content-Type`" go? https://docs.google.com/document/d/1jMJ4NxzHiBroNerUbiNWn5H01kgaPRvfRELr1WDvPRY/edit#heading=h.sug1aduexg87 is not specific enough as this is a JATS-specific constraint (but it could apply in general to any format that references external URIs of assets that need to be downloaded)

Reply:
Same as https://github.com/libero/content-store/pull/9#discussion_r282419676. Can guess the content type if it's not set, but for now this is part of the readme for this app.

Reply:
The readme looks like a good place.